### PR TITLE
feat(telemetry): report connector name and table_id along with meta

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

as title

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/meta/src/controller/fragment.rs [hunk 1]
```diff
@@ -655,6 +658,33 @@ impl CatalogController {
...
+    pub async fn list_stream_job_desc(&self) -> MetaResult<Vec<MetaTelemetryJobDesc>> {
+        let inner = self.inner.read().await;
+        let info: Vec<(TableId, Property)> = Table::find()
+            .select_only()
+            .column(table::Column::TableId)
+            .column(source::Column::WithProperties)
+            .join(JoinType::InnerJoin, table::Relation::Source.def())
+            .into_tuple()
+            .all(&inner.db)
+            .await?;
```
Comment (yezizp2012) [inline src/meta/src/controller/fragment.rs:670]:
```suggestion
        let info: Vec<(TableId, Option<Property>)> = Table::find()
            .select_only()
            .column(table::Column::TableId)
            .column(source::Column::WithProperties)
            .join(JoinType::LeftJoin, table::Relation::Source.def())
            .filter(table::Column::TableType.ne(TableType::Internal))
            .into_tuple()
            .all(&inner.db)
            .await?;
```
If you want to return all streaming jobs, you have to use left join and filter out internal tables. Btw, I think it's better to move this function to catalog.rs.

File: src/meta/src/manager/metadata.rs [hunk 2]
```diff
@@ -684,6 +686,41 @@ impl MetadataManager {
...
     }
 
+    pub async fn list_stream_job_desc(&self) -> MetaResult<Vec<MetaTelemetryJobDesc>> {
+        match self {
+            MetadataManager::V1(mgr) => {
+                let tables = mgr.catalog_manager.list_tables().await;
+                let sources = mgr.catalog_manager.list_sources().await;
+
+                let mut res = Vec::with_capacity(tables.len());
+                for table_def in tables {
```
Comment (yezizp2012) [inline src/meta/src/manager/metadata.rs:696]: Ditto, we need to filter out internal tables.

File: src/meta/src/manager/metadata.rs [hunk 3]
```diff
@@ -684,6 +686,41 @@ impl MetadataManager {
...
+                    if let Some(OptionalAssociatedSourceId::AssociatedSourceId(source_id)) =
+                        table_def.optional_associated_source_id
+                    {
+                        let source = sources.iter().find(|s| s.id == source_id);
+                        let connector_name = source
+                            .map(|s| s.with_properties.get("connector").map(|c| c.to_lowercase()))
+                            .unwrap()
```
[the remark this answers is not in the document]
Comment (tabVersion) [reply src/meta/src/manager/metadata.rs:703]: yes, we cannot create a source without specifying connector.

File: src/meta/src/manager/metadata.rs (hunk 2 above)
Comment (yezizp2012) [reply src/meta/src/manager/metadata.rs:696]: There's a field like table type in table catalog.

File: src/meta/src/controller/fragment.rs (hunk 1 above)
Comment (tabVersion) [reply src/meta/src/controller/fragment.rs]: fixed

File: src/meta/src/manager/metadata.rs [hunk 4]
```diff
@@ -684,6 +686,41 @@ impl MetadataManager {
...
+            MetadataManager::V1(mgr) => {
+                let tables = mgr.catalog_manager.list_tables().await;
+                let sources = mgr.catalog_manager.list_sources().await;
+
+                let mut res = Vec::with_capacity(tables.len());
+                for table_def in tables {
+                    if let Some(OptionalAssociatedSourceId::AssociatedSourceId(source_id)) =
+                        table_def.optional_associated_source_id
+                    {
+                        let source = sources.iter().find(|s| s.id == source_id);
```
[the remark this answers is not in the document]
Comment (tabVersion) [reply src/meta/src/manager/metadata.rs]: fixed
It is okay to tolerate the efficiency as long as the code just run once per 6 hours

**Review by TennyZhuang — APPROVED** [review]

File: src/meta/src/controller/catalog.rs [hunk 5]
```diff
@@ -2395,6 +2396,35 @@ impl CatalogController {
         Ok(version)
     }
 
+    pub async fn list_stream_job_desc_for_telemetry(
+        &self,
+    ) -> MetaResult<Vec<MetaTelemetryJobDesc>> {
+        let inner = self.inner.read().await;
+        let info: Vec<(TableId, Property)> = Table::find()
```
Comment (yezizp2012) [inline src/meta/src/controller/catalog.rs:2403]: This part seems to have been forgotten to be changed? And in catalog manager only table and mview have been selected, we should maintain logical consistency.

File: src/meta/src/controller/catalog.rs [hunk 6]
```diff
@@ -2395,6 +2396,40 @@ impl CatalogController {
...
```
Comment (yezizp2012) [inline src/meta/src/controller/catalog.rs:2403]:
```suggestion
        let info: Vec<(TableId, Option<Property>)> = Table::find()
```

**Review by yezizp2012 — APPROVED** [review]
LGTM

File: proto/telemetry.proto [hunk 7]
```diff
@@ -85,6 +85,19 @@ message MetaReport {
   // This field represents the "number of running streaming jobs"
   // and is used to indicate whether the cluster is active.
   uint32 stream_job_count = 5;
+  // stream_jobs is the list of running streaming jobs
+  // and is used to collect the table_id, connector_name and table_optimizations
+  repeated StreamJobDesc stream_jobs = 6;
+}
+
+enum TableOptimization {
```
Comment (st1page) [inline proto/telemetry.proto:93]: PlanOptimizations?

**Review by st1page — APPROVED** [review]

**Pull request merged**